src/Caller.cs

```diff
@@ -127,6 +127,36 @@ private IntPtr NativeHandle
         /// <returns>Returns the fuel consumed by the executing WebAssembly code or 0 if fuel consumption was not enabled.</returns>
         public ulong GetConsumedFuel() => ((IStore)this).Context.GetConsumedFuel();
 
+        /// <summary>
+        /// Gets the data from the Store's Context. 
+        /// </summary>
+        /// <returns></returns>
+        public object? GetData() => ((IStore)this).Context.GetData();
+        
+        /// <summary>
+        /// Gets the data from the Store's Context and attempts to cast it to the specified type. 
+        /// </summary>
+        /// <returns>The typed store data, or null in case of a failed cast</returns>
+        public T? GetData<T>() 
+            where T: class
+        {
+            var data = this.GetData();
+            if (data is null) { return null; }
+
+            try  
+            { 
+                return (T)data; 
+            }
+            catch (InvalidCastException _) 
+            { 
+                return null; 
+            }
+            catch
+            {
+                throw;
+            }
```

Comment:
I think you can just use the `as` operator here, to avoid throwing (and catching) the `InvalidCastException`, as it just returns `null` when the value cannot be cast to that type, which is what the above code effectively does:

```suggestion
            return data as T;
```

Reply:
Wasn't aware that as gives back null, that's handy. I think in that case GetData<T> doesn't do much anymore, so I removed it.